### ISSUE TITLE
Not clustering custom markers without coordinate prop to direct children

### ISSUE THREAD

**Issue by tguelenman** [venits/react-native-map-clustering#137]
Hi,
first of all thanks for your work.

I was a bit surprised to notice the following behavior. I have a MapScreen class, that renders Markers through another EventMarkerPin class (only showing parts of the code):

MapScreen:

```
class MapScreen extends Component<Props, State> {
  renderEventMarkerPins = () => {
    const {visibleEventIds} = this.props;
    return visibleEventIds.map(eventId => (
      <EventMarkerPin key={eventId} eventId={eventId} />
    ));
  };
  render() {
    return (
      <MapView>
        {this.renderEventMarkerPins()}
      </MapView>
    )
  }
}
```

EventMarkerPin:

```
class EventMarkerPin extends Component<Props> {
  render() {
    return (
      <Marker
        identifier={eventId}
        flat
        coordinate={coordinate}
        anchor={anchor}
        tracksViewChanges={false}
        onPress={this.openEvent}>
        {markerView}
      </Marker>
    );
  }
}
```

As you can see, I'm not passing any `coordinate` prop to EventMarkerPin, but I'm getting it with mapStateToProps:

```
const mapStateToProps = (state, ownProps) => {
  return {
    coordinate,
  };
};
export default connect(mapStateToProps)(withNavigation(EventMarkerPin));
```

Checking out your code, I found `isMarker` (https://github.com/venits/react-native-map-clustering/blob/master/lib/helpers.js) and realized, that I had to pass the coordinates to EventMarkerPin for it work.

It would be great if the package would support custom components as Markers that get their coordinates individually from the redux store, instead of having to pass them from the MapView level.

**Comment by Simofy** [comment]
@tguelenman There was similar discussion on #134

**Comment by venits** [comment]
Fixed in module version `3.2.0`

Thank you for your contribution ❤️

**Comment by tyler-coleman** [comment]
Out of curiosity, how was this fixed in 3.2.0?

It seems passing the coordinate prop directly to the custom marker is still necessary in order for clustering to work. Unfortunately, in my application, I'm fetching the markers' coordinates _in the function body of the custom marker itself_, and feeding the custom marker only an id which it uses to query our backend for the coordinate data. This is a necessary workaround for my particular use case.

Because I'm using a useQuery hook to fetch the coordinate for each map result, there really isn't anywhere higher in the component tree to perform that fetching because the results list will vary in length and I get RN errors for having varying numbers of hook calls between renders.
